radix-engine/src/vm/vm.rs

```diff
@@ -49,6 +52,68 @@ impl VmApi for VmBoot {
     }
 }
 
+/// The canonical implementation is [`VmModules`].
+pub trait VmInitialize {
```

Comment:
Not sure of the use of this trait? I don't think the we get anything by this abstraction?

Reply:
So the main use is that it encapsulates the generics (as associated types) to reduce the noise/pollution across the code base.

We can just take a parameter of impl VmInitialize instead of needing a generic parametrization over the native vm extension and scrypto vm.

But as that’s not clear, I’ll add a comment about it as a markup on a later PR 😄